src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -65,34 +66,98 @@ public override Task OnProposalUpdatedAsync(SuggestionSessionBase session, Propo
             return Task.CompletedTask;
         }
 
-        public async Task TryDisplaySuggestionAsync(CancellationToken cancellationToken)
+        /// <summary>
+        /// Starts the Suggestion Session. The TryDisplaySuggestion call doesn't display any grey text, but starts the session such that we have the
+        /// exclusive right to display grey text later.
+        /// </summary>
+        /// <returns>If true, user will see the thinking state as long as the Suggestion Session is active and replace with grey text if a call to DisplayProposal succeeds.
+        /// If unable to retrieve the session, the caller should bail out.
+        /// </returns>
+        public async Task<bool> StartSuggestionSessionAsync(CancellationToken cancellationToken)
         {
-            _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false);
+            _suggestionSession = await RunWithEnqueueActionAsync(
+                "StartWork",
+                async () => await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false),
+            cancellationToken).ConfigureAwait(false);
 
-            if (_suggestionSession != null)
+            if (_suggestionSession is null)
             {
-                var success = await TryDisplayProposalAsync(_suggestionSession, cancellationToken).ConfigureAwait(false);
-                if (success)
-                {
-                    Logger.Log(FunctionId.Copilot_Generate_Documentation_Displayed, logLevel: LogLevel.Information);
-                }
+                await DisposeAsync().ConfigureAwait(false);
```

Comment:
i def feel like this shouldbe commented to explain what's going on.  ou could even consider having a helper method with a clearer name that hten internallyc alls into DisposeAsync.